### PR TITLE
Add TypeScript and TSX parser

### PR DESCRIPTION
**Pull request by wadimw**

### CODE REVIEW

File: test/testfiles/TsxParser_TsFile.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import * as React from "react";
+import { Component, type CommonProps } from "components";
+
+interface WrappedComponentProps extends CommonProps {
+    count: number;
+}
+
+export class WrappedComponent extends React.Component<WrappedComponentProps> {
+    render() {
```
Comment (ehoogerbeets) [inline test/testfiles/TsxParser_TsFile.ts]: Need more types in here to verify that it is parsing typescript and not just jsx

File: test/testfiles/TsxParser_TsxFile.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import * as React from "react";
+import { Component, type CommonProps } from "components";
+
+interface WrappedComponentProps extends CommonProps {
+    count: number;
+}
+
+export class WrappedComponent extends React.Component<WrappedComponentProps> {
+    render() {
```
Comment (ehoogerbeets) [inline test/testfiles/TsxParser_TsxFile.tsx]: Same here

**Review by ehoogerbeets — APPROVED** [review]
This PR is okay to merge, so approving so you can merge after adding a few type constructs in the tests which I think won't really change any outcomes.

File: test/testfiles/TsxParser_TsFile.ts (hunk 1 above)
Comment (wadimw) [reply test/testfiles/TsxParser_TsFile.ts]: `import type`, `interface` and generic type annotation `<WrappedComponentProps>` are all TypeScript features though

Comment (wadimw) [reply test/testfiles/TsxParser_TsFile.ts]: Added explicit return type to the method in c684610

**Pull request merged**